Mathlib/Data/Finsupp/Fin.lean

```diff
@@ -93,8 +193,63 @@ lemma cons_support : (s.cons y).support ⊆ insert 0 (s.support.map (Fin.succEmb
   rintro i rfl
   simpa [Finsupp.mem_support_iff] using hi
 
-lemma cons_right_injective {n : ℕ} {M : Type*} [Zero M] (y : M) :
-    Injective (Finsupp.cons y : (Fin n →₀ M) → Fin (n + 1) →₀ M) :=
-  (equivFunOnFinite.symm.injective.comp ((Fin.cons_right_injective _).comp DFunLike.coe_injective))
+theorem cons_right_injective : Injective (Finsupp.cons y : (Fin n →₀ M) → Fin (n + 1) →₀ M) :=
+  equivFunOnFinite.symm.injective.comp ((Fin.cons_right_injective _).comp DFunLike.coe_injective)
+
+theorem snoc_ne_zero_of_left (h : s ≠ 0) : snoc s y ≠ 0 := by
+  contrapose! h with c
+  ext a
+  simp only [← snoc_castSucc y s a, c, coe_zero, Pi.zero_apply]
+
+theorem snoc_ne_zero_of_right (h : y ≠ 0) : snoc s y ≠ 0 := by
+  contrapose! h with c
+  rw [← snoc_last y s, c, Finsupp.coe_zero, Pi.zero_apply]
+
+theorem snoc_ne_zero_iff : snoc s y ≠ 0 ↔ s ≠ 0 ∨ y ≠ 0 := by
```

Comment:
Can you first prove `snoc_eq_zero`? The proof should be something like `by simp [funext_iff, Fin.forall_iff_succ]`